src/flags.rs

```diff
@@ -41,76 +42,84 @@ pub fn process(flags: &DenoFlags) {
---v8-options       Print V8 command line options.
---deps             Print module dependencies.
---types            Print runtime TypeScript declarations.
-
+pub fn get_usage(opts: &Options) -> String {
+  format!(
+    "Usage: deno script.ts {}
 Environment variables:
-DENO_DIR           Set deno's base directory."
-  );
+        DENO_DIR        Set deno's base directory.",
+    opts.usage("")
+  )
 }
 
 // Parses flags for deno. This does not do v8_set_flags() - call that separately.
 pub fn set_flags(
   args: Vec<String>,
-) -> Result<(DenoFlags, Vec<String>), String> {
+) -> Result<(DenoFlags, Vec<String>, String), String> {
   let args = v8_set_flags(args);
 
+  let mut opts = Options::new();
+  // TODO(kevinkassimo): v8_set_flags intercepts '-help' with single '-'
+  // Resolve that and then uncomment line below (enabling Go style -long-flag)
+  // opts.long_only(true);
+  opts.optflag("", "allow-write", "Allow file system write access.");
+  opts.optflag("", "allow-net", "Allow network access.");
+  opts.optflag("", "allow-env", "Allow environment access.");
+  opts.optflag("", "recompile", "Force recompilation of TypeScript code.");
+  opts.optflag("h", "help", "Print this message.");
+  opts.optflag("D", "log-debug", "Log debug output.");
+  opts.optflag("v", "version", "Print the version.");
+  opts.optflag("r", "reload", "Reload cached remote resources.");
+  opts.optflag("", "v8-options", "Print V8 command line options.");
+  opts.optflag("", "deps", "Print module dependencies.");
+  opts.optflag("", "types", "Print runtime TypeScript declarations.");
+
   let mut flags = DenoFlags::default();
-  let mut rest = Vec::new();
-  let mut arg_iter = args.iter();
 
-  while let Some(a) = arg_iter.next() {
-    if a.len() > 1 && &a[0..2] == "--" {
-      match a.as_str() {
-        "--help" => flags.help = true,
-        "--log-debug" => flags.log_debug = true,
-        "--version" => flags.version = true,
-        "--reload" => flags.reload = true,
-        "--recompile" => flags.recompile = true,
-        "--allow-write" => flags.allow_write = true,
-        "--allow-net" => flags.allow_net = true,
-        "--allow-env" => flags.allow_env = true,
-        "--deps" => flags.deps_flag = true,
-        "--types" => flags.types_flag = true,
-        "--" => break,
-        other => return Err(format!("bad option {}", other)),
-      }
-    } else if a.len() > 1 && &a[0..1] == "-" {
-      let mut iter = a.chars().skip(1); // skip the "-"
-      while let Some(f) = iter.next() {
-        match f {
-          'h' => flags.help = true,
-          'D' => flags.log_debug = true,
-          'v' => flags.version = true,
-          'r' => flags.reload = true,
-          other => return Err(format!("bad option -{}", other)),
-        }
-      }
-    } else {
-      rest.push(a.clone());
+  let matches = match opts.parse(&args) {
+    Ok(m) => m,
+    Err(f) => {
+      return Err(f.to_string());
     }
+  };
+
+  if matches.opt_present("help") {
```

Comment:
You would think that there would be some way to enumerate over the values, change the opts string from `foo-bar` to `foo_bar` and automatically set the flags.  It would have to be one less place to maintain when adding a flag then.

Reply:
Hmm I could look into that, though I don't think `getopts` crate itself offers some easy way to do this...

Reply:
I know you are already down this path a ways, but `docopt` seems like it would be a more direct way of getting a set of flags out.  See: https://docs.rs/docopt/1.0.1/docopt/#type-based-decoding